File: _more/compiler/py2rs/test/class_example.py
# Test with class and method

class Counter:
    def __init__(self):
        self.count = 0
    
    def increment(self):
        self.count += 1
        return self.count
    
    def get_count(self):
        return self.count

c = Counter()
c.increment()
c.increment()
print("count:", c.get_count())

File: _more/compiler/py2rs/test/class_example.rs
struct Counter {
    count: i32,
}

impl Counter {
    fn new() -> Counter {
        Counter { count: 0 }
    }

    fn increment(&mut self) -> i32 {
        self.count += 1;
        self.count
    }

    fn get_count(&self) -> i32 {
        self.count
    }
}

fn main() {
    let mut c = Counter::new();
    c.increment();
    c.increment();
    println!("count: {}", c.get_count());
}